docs/tutorials/tabular_prediction/tabular-custom-model.md

```diff
@@ -0,0 +1,374 @@
+
+In this example, we create a custom Random Forest model for use in AutoGluon. All models in AutoGluon inherit from the AbstractModel class ([AbstractModel source code](../../_modules/autogluon/core/models/abstract/abstract_model.html)), and must follow its API to work alongside other models.
+
+Note that while this tutorial provides a basic model implementation, this does not cover many aspects that are used in most implemented models.
+
+To best understand how to implement more advanced functionality, refer to the [source code](../../api/autogluon.tabular.models.html#module-autogluon.tabular.models) of the following models:
+
+- Respecting time limit / early stopping logic
+  - Refer to [LGBModel](../../_modules/autogluon/tabular/models/lgb/lgb_model.html#LGBModel) and [RandomForestModel](../../_modules/autogluon/tabular/models/rf/rf_model.html#RFModel)
+- Respecting memory usage limit
+  - Refer to LGBModel and RandomForestModel
+- Sample weight support
+  - Refer to LGBModel
+- Validation data and eval_metric usage
+  - Refer to LGBModel
+- GPU training support
+  - Refer to LGBModel
+- Save / load logic of non-serializable models
+  - Refer to [NNFastAiTabularModel](../../_modules/autogluon/tabular/models/fastainn/tabular_nn_fastai.html#NNFastAiTabularModel)
+- Advanced problem type support (Softclass, Quantile)
+  - Refer to RandomForestModel
+- Text feature type support
+  - Refer to [TextPredictorModel](../../_modules/autogluon/tabular/models/text_prediction/text_prediction_v1_model.html#TextPredictorModel)
+- Image feature type support
+  - Refer to [ImagePredictorModel](../../_modules/autogluon/tabular/models/image_prediction/image_predictor.html#ImagePredictorModel)
+- Custom HPO logic
+  - Refer to LGBModel
+
+## Implementing a custom model
+
+Here we define the custom model we will be working with for the rest of the tutorial.
+
+The most important methods that must be implemented are `_fit` and `_preprocess`.
+
+Follow along with the code comments to better understand how the code works.
+
+```{.python .input}
+import numpy as np
+import pandas as pd
+
+from autogluon.core.models import AbstractModel
+from autogluon.features.generators import LabelEncoderFeatureGenerator
+
+class CustomRandomForestModel(AbstractModel):
+    def __init__(self, **kwargs):
+        # Simply pass along kwargs to parent, and init our internal `_feature_generator` variable to None
+        super().__init__(**kwargs)
+        self._feature_generator = None
+
+    # The `_preprocess` method takes the input data and transforms it to the internal representation usable by the model.
+    # `_preprocess` is called by `preprocess` and is used during model fit and model inference.
+    def _preprocess(self, X: pd.DataFrame, is_train=False, **kwargs) -> np.ndarray:
+        print(f'Entering the `_preprocess` method: {len(X)} rows of data (is_train={is_train})')
```

Comment:
prints are spamming logs in the docs in later examples

Reply:
@jwmueller thoughts on this? The print statements are meant to help the user understand how and when the methods are called.

Reply:
I think these statements are very helpful. But if you really don't want them appear in later logs, you could give the model a hyperparameter that controls whether these are printed or not

Reply:
I'll plan to keep them for now, adding it as a hyperparameter would maybe detract from tutorial, but we can revisit later if we think controlling it is cleaner.